app/components/App.js

```diff
@@ -1,12 +1,33 @@
 import React from 'react';
+import Header from './Header';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      mostPopular: true,
+    };
+
+    this.onRefreshClick = this.onRefreshClick.bind(this);
+    this.onToggleClick = this.onToggleClick.bind(this);
+  }
+
+  onRefreshClick() {
+    // trigger get request to server '/load' route
   }
+
+  onToggleClick() {
+    this.setState({mostPopular: !this.state.mostPopular});
```

Comment:
Could be wrong on this, but I don't think `this.state` should be referenced inside a `setState` function call. Look into the callback version of `this.setState`